controllers/rollingupgrade_controller.go

```diff
@@ -244,28 +248,133 @@ func (r *RollingUpgradeReconciler) CallKubectlDrain(ctx context.Context, nodeNam
 	errChan <- nil
 }
 
-func (r *RollingUpgradeReconciler) WaitForTermination(nodeName string, nodeInterface v1.NodeInterface) (bool, error) {
+func (r *RollingUpgradeReconciler) WaitForDesiredInstances(ruObj *upgrademgrv1alpha1.RollingUpgrade) error {
+	var (
+		started = time.Now()
+	)
+	for {
+		if time.Since(started) >= (time.Second * time.Duration(ScaleoutTimeoutSeconds)) {
+			err := errors.New("WaitForDesired timed out while waiting for instance to be added")
+			return err
+		}
+		err := r.populateAsg(ruObj)
+		if err != nil {
+			return err
+		}
+		if getInServiceCount(r.Asg.Instances) == aws.Int64Value(r.Asg.DesiredCapacity) {
+			break
```

Comment:
Would be useful to have a log message here stating that the Desired count was reached in the ASG.